packages/fuels-signers/src/wallet.rs

```diff
@@ -407,8 +408,8 @@ impl Signer for Wallet {
         Ok(sig)
     }
 
-    fn address(&self) -> Address {
-        self.address
+    fn address(&self) -> &Bech32Address {
```

Comment:
Why are we returning a reference type now?

Reply:
`Address` implemented the `Copy` trait, so passing it to functions didn't transfer ownership. 
I opted for a reference to avoid copying memory.